src/systems/filecoin_vm/runtime/runtime.id

```diff
@@ -1,85 +1,81 @@
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
+import exitcode "github.com/filecoin-project/specs/systems/filecoin_vm/runtime/exitcode"
 import ipld "github.com/filecoin-project/specs/libraries/ipld"
-import st "github.com/filecoin-project/specs/systems/filecoin_vm/state_tree"
 import msg "github.com/filecoin-project/specs/systems/filecoin_vm/message"
-import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
-import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
+import st "github.com/filecoin-project/specs/systems/filecoin_vm/state_tree"
 
 // Randomness is a string of random bytes
 type Randomness Bytes
 
+type ActorStateHandle struct {
+    Take() actor.ActorSubstateCID
+    Release(checkValue actor.ActorSubstateCID)
+    UpdateRelease(newValue actor.ActorSubstateCID)
+
+    _initValue  actor.ActorSubstateCID?
+    _rt         Runtime
+}
+
 // Runtime is the VM's internal runtime object.
 // this is everything that is accessible to actors, beyond parameters.
 // FKA as vm.Context
 type Runtime struct {
-    // Invocation is the inputs for the current invocation
-    Invocation     InvocInput
-    State          VMState
-
-    // Fatal is a fatal error, and halts VM execution.
-    // This is the normal error condition of actor execution.
-    // On Fatal errors, the VM simply does not apply the state transition.
-    // This is atomic across the entire contract execution.
-    //
-    // TODO: @why wants to get rid of this.
-    Fatal(string)
+    CurrEpoch()              block.ChainEpoch
 
-    // Send allows the current execution context to invoke methods on other actors in the system
-    //
-    // TODO: this should change to async -- put the message on the queue.
-    //       do definied callback methods, with maybe a glue closure to align params or carry intermediate state.
-    //
-    // TODO: what are the return values here?
-    Send(
-        to      addr.Address
-        method  actor.MethodNum
-        params  actor.MethodParams
-        value   actor.TokenAmount
-    ) msg.MessageReceipt
-}
+    // Randomness returns a (pseudo)random stream (indexed by offset) for the current epoch.
+    Randomness(offset UInt)  Randomness
 
-// VMState is Chain state accessible to all contracts via the VM interface
-type VMState struct {
-    // StateTree returns the current state tree
-    StateTree  st.StateTree
+    Caller()                 addr.Address
+    ValidateCallerIs(caller addr.Address)
+    ValidateCallerMatches(CallerPattern)
 
-    // Storage provides access to the VM storage layer
-    Storage    VMStorage
+    AcquireState() ActorStateHandle
 
-    // BlockHeight returns the height of the block this message was added to the chain in
-    Epoch()    st.Epoch
+    ReturnSuccess()
+    ReturnValue(Bytes)
+    ReturnError(exitCode exitcode.ExitCode)
```

Comment:
- Why are we using these and not using return values in `ActorCode_I` functions as we used to?
- this is harming readability, and defeating/not using the go compiler's return value checks

Reply:
If anything, we can use helpers to make the types:

```go
func (a *CronActorCode_I) Constructor(rt vmr.Runtime) vmr.IOutput {
  ...
  return vmr.ReturnSuccess() // returns a &vmr.IOutput_I{ ... }
}
```

Reply:
Fixed. (I believe this was leftover from a draft in which each method was run in its own goroutine, and we needed a way to signal completion -- but this should no longer be relevant.)